src/public/app/widgets/llm_chat_panel.ts

```diff
@@ -0,0 +1,1127 @@
+            }
+        }
+
+        // Process the user message
+        await this.processUserMessage(content);
+
+        // Clear input and show loading state
+        this.noteContextChatInput.value = '';
+        this.showLoadingIndicator();
+        this.hideSources();
+
+        try {
+            const useAdvancedContext = this.useAdvancedContextCheckbox.checked;
+            const showThinking = this.showThinkingCheckbox.checked;
+
+            // Add logging to verify parameters
+            console.log(`Sending message with: useAdvancedContext=${useAdvancedContext}, showThinking=${showThinking}, noteId=${this.currentNoteId}, sessionId=${this.sessionId}`);
+
+            // Create the message parameters
+            const messageParams = {
+                content,
+                contextNoteId: this.currentNoteId,
+                useAdvancedContext,
+                showThinking
+            };
+
+            // First try to use streaming (preferred method)
+            try {
+                await this.setupStreamingResponse(messageParams);
+            } catch (streamingError) {
+                console.warn("Streaming request failed, falling back to direct response:", streamingError);
+
+                // If streaming fails, fall back to direct response
+                const handled = await this.handleDirectResponse(messageParams);
+                if (!handled) {
+                    // If neither method works, show an error
+                    throw new Error("Failed to get response from server");
+                }
+            }
+        } catch (error) {
+            this.handleError(error as Error);
+        }
+    }
+
+    /**
+     * Process a new user message - add to UI and save
+     */
+    private async processUserMessage(content: string) {
+        // Add user message to the chat UI
+        this.addMessageToChat('user', content);
+
+        // Add to our local message array too
+        this.messages.push({
+            role: 'user',
+            content,
+            timestamp: new Date()
+        });
+
+        // Save to note
+        this.saveCurrentData().catch(err => {
+            console.error("Failed to save user message to note:", err);
+        });
+    }
+
+    /**
+     * Try to get a direct response from the server
+     * @returns true if response was handled, false if streaming should be used
+     */
+    private async handleDirectResponse(messageParams: any): Promise<boolean> {
+        try {
+            // Add format parameter to maintain consistency with the streaming GET request
+            // This tells the server that we're expecting to use streaming even though
+            // we're sending a POST request
+            const postParams = {
+                ...messageParams,
+                format: 'stream'  // Match the format parameter used in the GET streaming request
+            };
+
+            console.log(`Sending direct POST request with params:`, postParams);
```

Comment:
bit of a nag comment: 
why not use template string here, when you are already use backticks 
```
console.log(`Sending direct POST request with params: ${postParams}`);
```